index.js

```diff
@@ -24,11 +25,15 @@ module.exports = {
     this.blacklist = this.generateBlacklist(config);
   },
 
-  treeForAddon: function() {
-    // see: https://github.com/ember-cli/ember-cli/issues/4463
-    var tree = this._super.treeForAddon.apply(this, arguments);
-    var moduleRegexp = '(^modules\/)?' + this.name + '\/helpers\/';
-    return this.filterHelpers(tree, new RegExp(moduleRegexp, 'i'));
+  treeForAddon: function(tree) {
+    tree = this.filterHelpers(tree, new RegExp(/^helpers\//, 'i'));
+    tree = new StripBadReexports(tree, [`index.js`]);
+    return this._super.treeForAddon.call(this, tree);
```

Comment:
Life is easier if you process the tree _before_ it has run through `_super.treeForAddon`. For one thing, you no longer need to worry about whether your current version of ember-cli includes the `modules` subdir (which is why the original RegExp was more complex). For another, we want our StripBadReexports plugin to process the import statements before they have been compiled into AMD by ember-cli-babel.